### PR TITLE
[vSphere] Add vSphere package

### LINKED ISSUE #642: Create vSphere package
# Integration release checklist

This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes

- [ ] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [ ] Supported versions of the monitoring target are documented
- [ ] Supported operating systems are documented (if applicable)
- [ ] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [ ] Documentation exists
- [ ] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [ ] At least a manual test with ES / Kibana / Agent has been performed.
- [ ] Required Kibana version set to:

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Adds initial vSphere package (copied from Metricbeat module).  Added syslog input for vCenter logs.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Author's Checklist

- [ ]

## How to test this PR locally
```
cd integrations/packages/pfsense
elastic-package check && elastic-package stack down && elastic-package build && elastic-package stack up -d -v && eval "$(elastic-package stack shellinit)" && elastic-package test  -v
```

## Related issues

- Closes #642 

## Screenshots
Dashboards copied from Metricbeat module
metricbeat_vsphere_vm_dashboard
metricbeat_vsphere_dashboard

### CODE REVIEW

File: packages/vsphere/_dev/build/build.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+dependencies:
+  ecs:
+    reference: git@1.11
```
Comment (legoguy1000) [inline packages/vsphere/_dev/build/build.yml:3]: Need to update to 1.12

File: packages/vsphere/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+---
+description: Pipeline for processing sample logs
+processors:  
+  - set:
+      field: event.ingested
+      value: '{{_ingest.timestamp}}'
+  - set:
+      field: ecs.version
+      value: '1.11.0'
```
Comment (legoguy1000) [inline packages/vsphere/data_stream/log/elasticsearch/ingest_pipeline/default.yml:9]: Need to update to 1.12

File: packages/vsphere/data_stream/virtualmachine/fields/ecs.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  type: keyword
+- description: Service address
+  name: service.address
+  type: keyword
+- description: Service type
+  name: service.type
+  type: keyword
+- description: Error message.
+  name: error.message
+  type: text
```
Comment (jsoriano) [inline packages/vsphere/data_stream/virtualmachine/fields/ecs.yml:12]: Use `external`.

File: packages/vsphere/manifest.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+name: vsphere
+title: VMware vSphere
+version: 0.0.1
+license: basic
+description: This Elastic integration collects logs from vSphere/vCenter servers
+type: integration
+categories:
+  - monitoring
+  - os_system
+release: experimental
```
Comment (jsoriano) [inline packages/vsphere/manifest.yml:11]:
```suggestion
release: beta
```

**Review by jsoriano — CHANGES_REQUESTED** [review]
Thanks for working on the migration of this integration!

**Comment by legoguy1000** [comment]
@jsoriano Updates made

**Review by jsoriano — APPROVED** [review]
Great, thanks!

**Comment by jsoriano** [comment]
@legoguy1000 there seems to be some formatting issues, can you take a look?

**Comment by legoguy1000** [comment]
> @legoguy1000 there seems to be some formatting issues, can you take a look?

Done.  The formatting will be the death of me

**Review by jsoriano — APPROVED** [review]

**Comment by jsoriano** [comment]
> do you know if this package is stuck in staging? Don't see it listed in our integration docs or within the Kibana integrations page (7.16).

Yes, I think this package has been never promoted to production. Pinging @elastic/integrations to see if this can be promoted.

**Comment by kaiyan-sheng** [comment]
I haven't tested this integration yet but seems like @marc-gr already made a bug fix on it. Marc, do you think this package is ready to be promoted? Also just a thought, maybe the owner should be the security-external-integrations team for now? 😄

**Pull request merged**